src/hazelcore/semantics/CursorInfo.re

```diff
@@ -946,6 +1049,101 @@ let ana_cursor_found_exp =
     );
   };
 
+let rec find_uses_block =
```

Comment:
Let's move the find_uses functions and the definition of uses_list into UHExp.re.

Reply:
I realized it may not work because the definition of `uses_list` depends on `CursorPath` while some definition in `CursorPath` depends on `UHExp`. And there would be a dependency cycle. How about changing `uses_list` to `CursorPath.steps_list`?